src/mpid/ch4/src/ch4_progress.h

```diff
@@ -13,18 +13,31 @@
 
 #include "ch4_impl.h"
 
+#define MPID_CH4_PROGRESS_NBC  (1)
+#define MPID_CH4_PROGRESS_NM   (1<<1)
+#define MPID_CH4_PROGRESS_SHM  (1<<2)
+
+#define MPID_CH4_PROGRESS_ALL (MPID_CH4_PROGRESS_NBC|MPID_CH4_PROGRESS_NM|MPID_CH4_PROGRESS_SHM)
+
+/* Flags argument allows to control execution of different parts of progress function,
```

Comment:
@raffenet `MPID_Progress_test_impl` is updated with short guidelines regarding reentrant-safety.